rdf-delta-server-local/src/main/java/org/seaborne/delta/server/local/patchstores/zk/PatchStoreZk.java

```diff
@@ -18,101 +18,82 @@
 
 import static org.seaborne.delta.server.local.patchstores.zk.ZkConst.*;
-import static org.seaborne.delta.zk.Zk.*;
 
-import java.io.ByteArrayOutputStream;
 import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Consumer;
 import java.util.regex.Pattern;
-import java.util.stream.Stream;
 
-import org.apache.curator.framework.CuratorFramework;
-import org.apache.curator.framework.recipes.locks.InterProcessLock;
 import org.apache.curator.utils.ZKPaths;
-import org.apache.jena.atlas.json.JSON;
 import org.apache.jena.atlas.json.JsonObject;
-import org.apache.jena.atlas.json.JsonValue;
-import org.apache.jena.atlas.lib.ListUtils;
 import org.apache.jena.atlas.lib.SetUtils;
 import org.apache.jena.atlas.logging.FmtLog;
 import org.apache.jena.atlas.logging.Log;
-import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.Watcher;
 import org.seaborne.delta.DataSourceDescription;
 import org.seaborne.delta.DeltaBadRequestException;
-import org.seaborne.delta.DeltaException;
 import org.seaborne.delta.server.local.LocalServerConfig;
 import org.seaborne.delta.server.local.PatchLog;
 import org.seaborne.delta.server.local.PatchStore;
 import org.seaborne.delta.server.local.PatchStoreProvider;
 import org.seaborne.delta.server.local.patchstores.PatchLogIndex;
 import org.seaborne.delta.server.local.patchstores.PatchStorage;
-import org.seaborne.delta.zk.Zk;
-import org.seaborne.delta.zk.Zk.ZkRunnable;
-import org.seaborne.delta.zk.Zk.ZkSupplier;
+import org.seaborne.delta.zk.UncheckedZkConnection;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class PatchStoreZk extends PatchStore {
     private final static Logger LOGZK = LoggerFactory.getLogger(PatchStoreZk.class);
-    // Use ServiceLoader.
-    public static final String PatchStoreZkName = "PatchStoreZk";
     // Schema!
     // https://curator.apache.org/curator-framework/schema.html
 
     // ---- Watching for logs
     private final Object storeLock = new Object();
 
-    private static AtomicInteger counter = new AtomicInteger(1);
-    private final CuratorFramework client;
-    private final InterProcessLock zkStoreLock;
+    private static final AtomicInteger counter = new AtomicInteger(1);
+    private final UncheckedZkConnection client;
     private final String instance;
 
     // Our view of all the patch logs in the store.
-    private Map<String, PatchLog> patchLogs = new ConcurrentHashMap<>();
+    private final Map<String, PatchLog> patchLogs = new ConcurrentHashMap<>();
 
-    protected PatchStoreZk(CuratorFramework client, PatchStoreProvider psp) {
+    protected PatchStoreZk(UncheckedZkConnection client, PatchStoreProvider psp) {
         super(psp);
         this.instance = "zk-"+(counter.getAndIncrement());
         this.client = client;
-        zkStoreLock = Zk.zkCreateLock(client, ZkConst.pStoreLock);
```

Comment:
I deleted this because it isn't used anyway, however, now that I'm reviewing it, this appears to be related to concurrency. Your integration tests passed despite this change, however, I don't think any of your integration tests check concurrency. I will be adding concurrency tests both to prove to you that the Curator version does not work and because we need it to have confidence that this application performs as expected in the desired use-case. We will uncover in the tests if this lock is truly necessary.